### PR TITLE
KAFKA-12562: Remove deprecated APIs in KafkaStreams and returned state classes

### PR DESCRIPTION
**Pull request by guozhangwang**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -98,6 +98,9 @@ <h3><a id="streams_api_changes_300" href="#streams_api_changes_300">Streams API
     </p>
     <ul>
         <li> <code>WindowBytesStoreSupplier#segments</code>: deprecated in Kafka 2.1.0 (<a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-319%3A+Replace+segments+with+segmentInterval+in+WindowBytesStoreSupplier">KIP-319</a>).</li>
+        <li> Overloaded <code>KafkaStreams#close</code>: deprecated in Kafka 2.1.0 (<a href="https://cwiki.apache.org/confluence/display/KAFKA/KIP-319%3A+Replace+segments+with+segmentInterval+in+WindowBytesStoreSupplier">KIP-319</a>).</li>
```
Comment (ableegoldman) [inline docs/streams/upgrade-guide.html]: Missing note for removed methods on KeyQueryMetadata and StreamsMetadataState

Comment (guozhangwang) [reply docs/streams/upgrade-guide.html]: KeyQueryMetadata: ah yes, will add.
StreamsMetadataState: it is the internal class.

Comment (guozhangwang) [reply docs/streams/upgrade-guide.html]: EDIT: deprecated APIs in KeyQueryMetadata are part of KIP-648 (2.7) so should not be removed, will add back.

**Comment by guozhangwang** [comment]
@ableegoldman Thanks for the reviews. I will merge after green builds.

**Comment by guozhangwang** [comment]
Bumped into https://issues.apache.org/jira/browse/KAFKA-12557, there's a PR fixing it already.

**Pull request merged**